lib/src/layer/polygon_layer/polygon_layer.dart

```diff
@@ -118,16 +127,37 @@ class _PolygonLayerState extends State<PolygonLayer> {
     );
   }
 
-  List<Polygon> _computeZoomLevelSimplification(int zoom) =>
-      _cachedSimplifiedPolygons[zoom] ??= widget.polygons
-          .map(
-            (polygon) => polygon.copyWithNewPoints(
-              simplify(
-                polygon.points,
-                widget.simplificationTolerance / math.pow(2, zoom),
-                highestQuality: true,
-              ),
+  static List<_ProjectedPolygon> _computeZoomLevelSimplification(
+    List<_ProjectedPolygon> polygons,
+    double tolerance,
+    int zoom,
+  ) =>
+      List<_ProjectedPolygon>.generate(
+        polygons.length,
+        (i) {
+          final polygon = polygons[i];
+          final holes = polygon.holePoints;
+
+          return _ProjectedPolygon._(
```

Comment:
Does this not mean that (if simplification is enabled), every polygon is converted to a `_ProjectedPolygon` twice (once prior to simplification then again in simplification), and all the expensive loops that means?
Wouldn't it be better to perform simplification at the same time as the first conversion, so two sets of loops don't need to be done?

Reply:
> That said, I do believe it's more correct to simplify projected polygons than it is to simplify polygons. Calculating all the simplifications and distances on a spherical coordinates pre-projection is kind of a recipe for projection weirdness. Are two points that are close on the surface of a sphere still close after projecting them onto the screen? (Remember, I initially had some concerns with the simplifications around the poles, simplifying projected points solves that issue... and yes we should change polylines as well)

Yep, 100% agreed with this. Need to do the same with triangulation as well.

However, we do one loop and one double-nested loop per conversion. We convert twice, because we convert to projected, then convert projected to projected via simplification. Would it not be better if we simplify within the loops of the initial projection, to save one conversion cycle?

Reply:
> However, we do one loop and one double-nested loop per conversion. We convert twice, because we convert to projected, then convert projected to projected via simplification. Would it not be better if we simplify within the loops of the initial projection, to save one conversion cycle?

There are many simplifications, one per zoom level. If you simplify for one zoom level during projection, you have to reproject again for another zoom level.